api/base/utils.py

```diff
@@ -194,9 +194,11 @@ def is_deprecated(request_version, min_version=None, max_version=None):
     return False
 
 
-def waterbutler_api_url_for(node_id, provider, path='/', _internal=False, **kwargs):
+def waterbutler_api_url_for(waterbutler_url, node_id, provider, path='/', _internal=False, **kwargs):
```

Comment:
Currently experimenting with passing in waterbutler_url first argument.  Not sure if quickfiles have a region (would assume it would be user settings region).

Reply:
+1 for saving a query and passing in the waterbutler base url.

I do think it should be added to the end of the arguments so as to not break existing usages and to avoid the awkward `waterbutler_api_url_for(None, ...)`. 

Also, the naming could be misleading. Maybe `base_url`?

So...

```python
def waterbutler_api_url_for(node_id, provider, path='/', _internal=False, base_url=None, **kwargs):
```